.github/workflows/python-package.yml

```diff
@@ -27,18 +27,23 @@ jobs:
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
-        python -m pip install flake8 pytest
+        python -m pip install pytest pre-commit
         if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
```

Comment:
After this line, please add

```
+        git clone https://gitlab.com/vaishakp/waveformtools.git
+        cd waveformtools
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+        export PYTHONPATH=${PWD}:${PYTHONPATH}
```

This will install a missing dependency that is needed by `pytest` below. Having said this, @vaishakp using `PYTHONPATH` is now archaic :)

Reply:
I should just be able to do `pip install .` instead of `export PYTHONPATH=${PWD}:${PYTHONPATH}` right?

Reply:
This is for @vaishakp to say.

Reply:
It does install locally:

```
$ git clone https://gitlab.com/vaishakp/waveformtools.git && cd waveformtools && pip install .
Cloning into 'waveformtools'...
remote: Enumerating objects: 1913, done.
remote: Counting objects: 100% (295/295), done.
remote: Compressing objects: 100% (292/292), done.
remote: Total 1913 (delta 179), reused 0 (delta 0), pack-reused 1618
Receiving objects: 100% (1913/1913), 9.75 MiB | 1012.00 KiB/s, done.
Resolving deltas: 100% (1231/1231), done.
Processing /home/prayush/src/waveformtools
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: waveformtools
  Building wheel for waveformtools (pyproject.toml) ... done
  Created wheel for waveformtools: filename=waveformtools-0.0.1-py3-none-any.whl size=84873 sha256=975a044f83eac29a39b1114c3c00c50f24c0d5f9e838c0cabe4c711949d24260
  Stored in directory: /tmp/pip-ephem-wheel-cache-4ombvzzw/wheels/56/48/a4/aa68ab93cd4c8f3b2288bb877a27e16aaa72b4d8d37d8c39aa
Successfully built waveformtools
Installing collected packages: waveformtools
Successfully installed waveformtools-0.0.1

```

and `pytest` runs successfully afterward:
```
$ unset PYTHONPATH
$ pytest
================================================ test session starts =================================================
platform linux -- Python 3.10.10, pytest-7.2.2, pluggy-1.0.0
rootdir: /home/prayush/src/nr-catalog-tools
plugins: anyio-3.6.2
collected 1 item                                                                                                     

test/test_sxs_modes.py .                                                                                       [100%]

================================================== warnings summary ==================================================
../../miniconda3/envs/lalsuite-dev-sxs/lib/python3.10/zipfile.py:1517
  /home/prayush/miniconda3/envs/lalsuite-dev-sxs/lib/python3.10/zipfile.py:1517: UserWarning: Duplicate name: 'catalog.json'
    return self._open_to_write(zinfo, force_zip64=force_zip64)

../../miniconda3/envs/lalsuite-dev-sxs/lib/python3.10/site-packages/sxs/waveforms/nrar.py:176
  /home/prayush/miniconda3/envs/lalsuite-dev-sxs/lib/python3.10/site-packages/sxs/waveforms/nrar.py:176: UserWarning: 
  Could not find root group as `h5_group` or as `extrapolation_order`; returning all groups
    warnings.warn(warning)

../../miniconda3/envs/lalsuite-dev-sxs/lib/python3.10/site-packages/waveformtools/dataIO.py:1007
  /home/prayush/miniconda3/envs/lalsuite-dev-sxs/lib/python3.10/site-packages/waveformtools/dataIO.py:1007: FutureWarning: Unlike other reduction functions (e.g. `skew`, `kurtosis`), the default behavior of `mode` typically preserves the axis it acts along. In SciPy 1.11.0, this behavior will change: the default value of `keepdims` will become False, the `axis` over which the statistic is taken will be eliminated, and the value None will no longer be accepted. Set `keepdims` to True or False to avoid this warning.
    dt_auto = mode(np.diff(time_axis))[0][0]

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================================== 1 passed, 3 warnings in 20.06s ===========================================
```
so I think its safe to `pip install .`.

Reply:
Yup, I tried it on CI as well. In fact, `pip install git+https://gitlab.com/vaishakp/waveformtools` also works. I think at this point we can also add this to `requirements.txt` or `optional_requirements.txt`, but of course this is not necessary.

Reply:
OK @vaishakp right now `pip install git+https://gitlab.com/vaishakp/waveformtools` doesn't work very well because the `setup.py` in your repository does not automatically install requirements. It seems that I cannot open a MR/issue there and cannot even fork the project. But can you just read in the requirements.txt file using 

```python
with open(fname, "r") as ff:
    requirements = ff.readlines()

```

and then pass `install_requires=requirements` as an argument to `setuptools.setup()`?

Reply:
If you'd rather not do this, then I will just go back to cloning, and then installing requirements explicitly. Please let me know 😄

Reply:
Re

>  I think at this point we can also add this to `requirements.txt` or `optional_requirements.txt`, but of course this is not necessary.

Since `waveformtools` is only used in tests, it hasnt been made a requirement yet. I have just opened #23 , but will wait for @vaishakp to okay on this.

Reply:
```
wget https://gitlab.com/vaishakp/waveformtools/-/raw/main/requirements.txt && pip install -r requirements.txt
pip install git+https://gitlab.com/vaishakp/waveformtools
```

For now, @adivijaykumar you can probably go with this ^, and remove the first line in time.